### PR TITLE
[autolinking] support multiple podspec in a package

### PR DESCRIPTION
**Pull request by Kudo**
# Why

ios autolinking enhancement to support multiple podspecs in one package

# How

resolve as `pods: [ { podName: 'podName', podspecDir: '/path/to/podDir' } ]` 

# Test Plan

- ci passed
- manually add a podspec into expo-modules-core and see if autolinking to add the pod

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-modules-autolinking/src/types.ts [hunk 1]
```diff
@@ -35,7 +35,15 @@ export type SearchResults = {
...
-export type ModuleDescriptor = Record<string, any>;
+export type ModuleDescriptorAndroid = Record<string, any>;
+export interface ModuleDescriptorIos {
+  pods: { podName: string; podspecDir: string }[];
+  flags: Record<string, any> | undefined;
+  modulesClassNames: string[];
+  appDelegateSubscribers: string[];
+}
+
+export type ModuleDescriptor = ModuleDescriptorAndroid | ModuleDescriptorIos;
```
Comment (tsapeta) [inline packages/expo-modules-autolinking/src/types.ts:46]: Thanks for improving this 🤗

**Review by tsapeta — APPROVED** [review]
Looks good, great work! 👍 I'm happy we don't have to do maintain backwards compatibility here 😉

**Comment by brentvatne** [comment]
it looks like this needs to be rebased. should we be trying to land this for sdk 44?

**Comment by Kudo** [comment]
this can be on next sdk for another attempt of expo-modules-core prebuilding 🤨

**Comment by Kudo** [comment]
close in favor of #16511

**Pull request closed without merging**